pkg/repository/git.go

```diff
@@ -32,27 +34,34 @@ func newGitRepoController() gitRepoController {
 }
 
 // IsClean checks if the repository have changes over the commit
-func (r gitRepoController) isClean() (bool, error) {
-	repo, err := r.repoGetter.get(r.path)
-	if err != nil {
-		return false, fmt.Errorf("failed to analyze git repo: %w", err)
-	}
-	worktree, err := repo.Worktree()
-	if err != nil {
-		return false, fmt.Errorf("failed to infer the git repo's current branch: %w", err)
+func (r gitRepoController) isClean(ctx context.Context) (bool, error) {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Second)
```

Comment:
Fixes #2970

Reply:
we are avoiding getting stuck but we should investigate the reason to face this timeout I think